daemon/stats.go

```diff
@@ -146,17 +156,37 @@ func (daemon *Daemon) unsubscribeToContainerStats(c *container.Container, ch cha
 
 // GetContainerStats collects all the stats published by a container
 func (daemon *Daemon) GetContainerStats(container *container.Container) (*types.StatsJSON, error) {
-	stats, err := daemon.stats(container)
-	if err != nil {
-		return nil, err
-	}
+	var stats *types.StatsJSON
+	var err error
+
+	stats, err = daemon.stats(container)
 
 	// We already have the network stats on Windows directly from HCS.
-	if !container.Config.NetworkDisabled && runtime.GOOS != "windows" {
-		if stats.Networks, err = daemon.getNetworkStats(container); err != nil {
+	if err == nil && !container.Config.NetworkDisabled && runtime.GOOS != "windows" {
+		stats.Networks, err = daemon.getNetworkStats(container)
+	}
+
+	switch err.(type) {
+	case nil:
+		// Sample system CPU usage close to container usage to avoid
+		// noise in metric calculations.
+		systemUsage, onlineCPUs, err := getSystemCPUUsage()
```

Comment:
@corhere I'm not clear about this. Could you suggest the order?

Reply:
Container usage (1) is sampled with `daemon.stats(container)`. System CPU usage (3) is sampled with `getSystemCPUUsage()`.  In between those calls is a call to `daemon.getNetworkStats(container)`. "Sample system CPU usage close to container usage to avoid noise in metric calculations."

Reply:
@corhere I pushed a new [commit](https://github.com/moby/moby/commit/3d70af4ee46874fbc859878f3f22ee4f6aa5f5f3) today for this minor change. Please check if it is what you want. Thanks.